mock/py/mockbuild/backend.py

```diff
@@ -509,34 +509,47 @@ def rebuild_package(self, spec_path, timeout, check):
         additional_opts = [self.config.get('rpmbuild_opts', '')]
         if additional_opts == ['']:
             additional_opts = []
+
+        def get_command(mode):
+            command = [self.config['rpmbuild_command']] + mode + \
+                      ['--target', self.rpmbuild_arch, '--nodeps'] + \
+                      check_opt + [spec_path] + additional_opts
+            command = ["bash", "--login", "-c"] + [' '.join(command)]
+            return command
+
         bd_out = self.make_chroot_path(self.buildroot.builddir)
-        command = [self.config['rpmbuild_command']] + mode + \
-                  ['--target', self.rpmbuild_arch, '--nodeps'] + \
-                  check_opt + [spec_path] + additional_opts
-        command = ["bash", "--login", "-c"] + [' '.join(command)]
-        try:
-            self.buildroot.doChroot(command,
-                                    shell=False, logger=self.buildroot.build_log, timeout=timeout,
-                                    uid=self.buildroot.chrootuid, gid=self.buildroot.chrootgid,
-                                    user=self.buildroot.chrootuser,
-                                    nspawn_args=self._get_nspawn_args(),
-                                    unshare_net=self.private_network,
-                                    printOutput=self.config['print_main_output'])
-        except Error as e:
-            if e.resultcode != 11:
-                raise e
-            self.buildroot.root_log.info("Dynamic buildrequires detected")
-            buildreqs = glob.glob(bd_out + '/SRPMS/*.buildreqs.nosrc.rpm')
-            self.installSrpmDeps(*buildreqs)
-            for f_buildreqs in buildreqs:
-                os.remove(f_buildreqs)
-            self.buildroot.doChroot(command,
-                                    shell=False, logger=self.buildroot.build_log, timeout=timeout,
-                                    uid=self.buildroot.chrootuid, gid=self.buildroot.chrootgid,
-                                    user=self.buildroot.chrootuser,
-                                    nspawn_args=self._get_nspawn_args(),
-                                    unshare_net=self.private_network,
-                                    printOutput=self.config['print_main_output'])
+        if self.config.get('dynamic_buildrequires'):
+            # Technically, we should run rpmbuild+installSrpmDeps until
+            # * it fails
+            # * installSrpmDeps does nothing
+            # but we don't have necessary infrastructure in place
```

Comment:
What does this mean?

Reply:
we need to know how many packages `installSrpmDeps` installed while right now everything what it does is just executes `dnf builddep`

Reply:
So in theory, we could call it a fixed number of times and document it, right? Like twice?

Reply:
I can imagine that we need to run it more times... and doing it just fixed number will make everything slow... I just need probably to inject rpm -qa | wc -l between... But I'd leave this part for somebody else to implement.

I'll happily review PR.

Reply:
I see. We can leave that open for future.